### ISSUE TITLE
OS compatability?

### ISSUE THREAD

**Issue by rorosaurus** [sonocotta/mouse-whisperer#1]
If this is based on the Attiny85 and Digispark’s DigiMouse.h, isn’t it susceptible to not working on many systems? I’ve seen lots of reports of it not working on: Windows, OSX, and Linux on hackaday.

I just built a Digistump mouse jiggler last night and couldn’t get it to be recognized by my Windows desktop (even with a USB hub), but it worked, albeit intermittently, on my old Macbook - though only sometimes unless a USB hub was used too.

https://hackaday.io/project/28445-usb-cdc-robosapien-v1/log/95249-legacy-getting-started-with-digistump-clone
https://hackaday.com/2018/09/17/diy-rubber-ducky-is-as-cheap-as-its-namesake/

Has this design been tested on these other configurations? Are you doing anything special to get the Attiny85 to work more reliably?

**Comment by anabolyc** [comment]
You're right, it is based on Digispark’s DigiMouse.h, and USB implementation is a bit shaky there with assembler code trying hard to comply with USB timings. I suspect it will not comply 100% of time, and some USB hosts will reject it. Better would be to use MCU with hardware USB support, like Microchip's Atmega32u4 or similar. However those are 4 times more expensive and beats the whole idea

I'm planning to work on modern Tiny port, which has much more stable internal clock. But so far this is as good as it gets - I tested device on few Linux hosts and few Windows - all work without an issue, although it will probably fail for some others.
